pkg/boot/fit/fit.go

```diff
@@ -49,14 +47,46 @@ func New(n string) (*Image, error) {
 	return &Image{name: n, Root: fdt}, nil
 }
 
+// ParseConfig reads r for a FIT image and returns a OSImage for each
+// configuration parsed.
+func ParseConfig(r io.ReadSeeker, verbose bool) ([]Image, error) {
+	fdt, err := dt.ReadFDT(r)
+	if err != nil {
+		return nil, err
+	}
+
+	var images []Image
+	configs := fdt.Root().Walk("configurations")
+	cn, _ := configs.ListChildNodes()
+
+	for _, n := range cn {
+		var i = Image{name: n, Root: fdt, ConfigOverride: n}
+
+		kn, in, err := i.LoadConfig(false, verbose)
+
+		if err == nil {
+			i.Kernel, i.InitRAMFS = kn, in
+			images = append(images, i)
+		}
+		if err != nil && verbose {
```

Comment:
just return the error. Don't do prints like this from a function. Also, take the verbose parameter out.

Reply:
I returned an Image for each FIT config, so the error wasn't fatal. I removed it and added a simpler msg if no valid imgs are found.